### PR TITLE
added new disabled prop and refactored previous defaultValue logic

### PR DESCRIPTION
**Pull request by MRArguello**
This PR includes a small refactor on the logic for `defaultValue` in order to keep the `null` values when needed (previous implementation was returning `undefined`).

It also adds support for `disable` prop and implements both things in the example forms and inputs.

### CODE REVIEW

File: src/components/FormBuilder/FormBuilder.tsx [hunk 1]
```diff
@@ -6,13 +6,12 @@ const FormBuilder: FormBuilderType<any> = (
   { fields, fieldOptions = {}, values, validate, handleSubmit, submitSection }) => {
 
   const setFormObject = (currentValues: any = {}) => {
-    const defaultValues = Object.keys(fieldOptions).reduce((defaultsObject, key) => {
-      const current = fieldOptions[key] || { defaultValue: null };
-
-      return { ...defaultsObject, [key]: current.defaultValue };
+    const defaults = fields.reduce((acc, field) => {
+      const options = fieldOptions[field] as FieldOptionsValueType<any>;
+      const { defaultValue } = options.defaultValue ?  options : { defaultValue: null };
```
Comment (erhangundogan) [inline src/components/FormBuilder/FormBuilder.tsx]: double space before `options`

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@savedo/re-form",
-  "version": "1.3.0",
+  "version": "1.4.0",
```
Comment (erhangundogan) [inline package.json]: it's bug fix so bumping patch version would be fine `1.3.1`

File: src/components/FormBuilder/FormBuilder.tsx [hunk 3]
```diff
@@ -6,13 +6,12 @@ const FormBuilder: FormBuilderType<any> = (
   { fields, fieldOptions = {}, values, validate, handleSubmit, submitSection }) => {
 
   const setFormObject = (currentValues: any = {}) => {
-    const defaultValues = Object.keys(fieldOptions).reduce((defaultsObject, key) => {
-      const current = fieldOptions[key] || { defaultValue: null };
-
-      return { ...defaultsObject, [key]: current.defaultValue };
+    const defaults = fields.reduce((acc, field) => {
+      const options = fieldOptions[field] as FieldOptionsValueType<any>;
```
Comment (michaelrevans) [inline src/components/FormBuilder/FormBuilder.tsx]: now that you're using `fieldOptions[field]`, you can simplify it in the way we tried earlier:

```suggestion
      const { defaultValue = null } = fieldOptions[field] as FieldOptionsValueType<any>;
```
and remove the following line

File: package.json (hunk 2 above)
Comment (erhangundogan) [reply package.json]: Sorry I missed new feature which is `disabled` prop. You can keep 1.4.0

File: src/components/FormBuilder/FormBuilder.tsx [hunk 4]
```diff
@@ -6,13 +6,12 @@ const FormBuilder: FormBuilderType<any> = (
   { fields, fieldOptions = {}, values, validate, handleSubmit, submitSection }) => {
 
   const setFormObject = (currentValues: any = {}) => {
-    const defaultValues = Object.keys(fieldOptions).reduce((defaultsObject, key) => {
-      const current = fieldOptions[key] || { defaultValue: null };
-
-      return { ...defaultsObject, [key]: current.defaultValue };
+    const defaults = fields.reduce((acc, field) => {
```
Comment (michaelrevans) [inline src/components/FormBuilder/FormBuilder.tsx]: I'm not as much a fan of `acc` - even though it only exists inside of here to denote the accumulated value, it's not very descriptive (as we discussed earlier)

**Pull request merged**